boards/armv7m/default_handlers.h

```diff
@@ -128,14 +128,29 @@ void hard_fault_handler_step_3(void);
 
 __attribute__((__naked__)) static void hard_fault_handler(void)
 {
+// set switch to 0 in order to alternatively recreate the previous stack frame
+// and hald the CPU
+#if 1
     __asm volatile
     (
         " tst   lr, #4               \n"
         " ite   eq                   \n"
         " mrseq r0, msp              \n"
         " mrsne r0, psp              \n"
-        " ldr r1, =hard_fault_handler_step_2 \n"
-        " bx r1 \n");
+        " ldr r1, [r0, #24]          \n"
```

Comment:
I don't understand this change. Where does this value go? hard fault handler only has one argument.

Reply:
I'm laying the ground work to merge in the SP restore later.